From b73a7fe7404b1ac7baa17e16d4f062f8f864de49 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:15 -0700
Subject: [PATCH 1640/1852] v4l: xilinx: vpss-csc: Fix warning for
 PW.MIXED_ENUM_TYPE

commit 6d20e2715ca1f6a812feca212b20f364f68b504a from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the PW.MIXED_ENUM_TYPE type.

The actual warnings are:

For xcsc_parse_of():
Event mixed_enum_type: 	enumerated type mixed with another type

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-csc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-csc.c b/drivers/media/platform/xilinx/xilinx-vpss-csc.c
index 8e6fd16e5630..51c7440bf617 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-csc.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-csc.c
@@ -1047,8 +1047,10 @@ static int xcsc_parse_of(struct xcsc_dev *xcsc)
 	}
 	switch (video_width[0]) {
 	case XVIDC_BPC_8:
+		xcsc->color_depth = XVIDC_BPC_8;
+		break;
 	case XVIDC_BPC_10:
-		xcsc->color_depth = video_width[0];
+		xcsc->color_depth = XVIDC_BPC_10;
 		break;
 	default:
 		dev_err(dev, "Unsupported color depth %d", video_width[0]);
-- 
2.31.1

